multiple_dc_test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2017 ScyllaDB
+
+import logging
+
+from avocado import main
+
+from sdcm.tester import ClusterTester
+
+
+class MultipleDcTest(ClusterTester):
+    """
+
+    :avocado: enable
+    """
+
+    def stop_scylla(self, node):
+        node.wait_db_up()
+        node.remoter.run('sudo systemctl stop scylla-server.service')
+        node.remoter.run('sudo systemctl stop scylla-jmx.service')
+        node.wait_db_down()
+
+    def start_scylla(self, node):
+        node.wait_db_down()
+        node.remoter.run('sudo systemctl start scylla-server.service')
+        node.remoter.run('sudo systemctl start scylla-jmx.service')
+        node.wait_db_up()
+
+    def test_shutdown(self):
+    """
+    https://github.com/scylladb/scylla-enterprise/issues/228
+    """
+
+        # run a background workload
+        stress_queue = self.run_stress_thread(stress_cmd=self.params.get('stress_cmd'),
+                                              stress_num=2,
+                                              keyspace_num=1)
+        nodes = self.db_cluster.nodes
+
+        # add a new node to existing cluster
+        new_node = self.add_node(1)[0]
+
+        # stop old node
+        old_node = nodes[-1]
+        self.stop_scylla(old_node)
+
+        # stop new node
+        self.stop_scylla(new_node)
+
+        # restart the service
+        self.start_scylla(old_node)
+        self.start_scylla(new_node)
+
+        results = self.get_stress_results(queue=stress_queue, stress_num=2, keyspace_num=1)
+
+    def test_series_stop(self):
+
+        # run a background workload
+        stress_queue = self.run_stress_thread(stress_cmd=self.params.get('stress_cmd'),
+                                              stress_num=2,
+                                              keyspace_num=1)
+        nodes_2nd_dc = [n for n in node_list if n.dc_idx == 1]
+        for node in nodes_2nd_dc:
+            self.stop_scylla(node)
+
+        for node in nodes_2nd_dc:
+            self.start_scylla(node)
+
+        results = self.get_stress_results(queue=stress_queue, stress_num=2, keyspace_num=1)
+
```

Comment:
I just checked get_stress_results(), if some c-s process doesn't return result summary for some reason(error), get_stress_results() won't raise exception and just return a null list as subresult.

I will add an option for get_stress_results, to check valid result insider it. And we can also do some complex checking outside get_stress_results().

Thanks for the suggestion!